src/PKSim.Core/Model/ParameterExtensions.cs

```diff
@@ -207,6 +207,23 @@ public static void ScaleDistributionBasedOn(this IDistributedParameter parameter
 
          parameter.IsFixedValue = false;
       }
+      
+      /// <summary>
+      /// Returns false if the proposed <paramref name="newValueInBaseUnit"/> will make the value of <paramref name="effectiveMolWeightParameter"/> negative
+      /// </summary>
+      public static bool IsEffectiveMolWeightPositive(this IParameter parameter, double newValueInBaseUnit, IParameter effectiveMolWeightParameter)
+      {
+         var oldValue = parameter.Value;
+         try
+         {
+            parameter.Value = newValueInBaseUnit;
+            return effectiveMolWeightParameter.Value > 0;
+         }
+         finally
+         {
+            parameter.Value = oldValue;
+         }
```

Comment:
Checking this requires evaluation of the effective mol weight formula. In this code I'm editing it in-place then reverting the old value in the finally.